org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/internal/SynchronizeGradleBuildOperation.java

```diff
@@ -219,8 +219,8 @@ private void synchronizeOpenWorkspaceProject(OmniEclipseProject project, IProjec
 
     private PersistentModel loadCurrentModel(IProject workspaceProject) {
         PersistentModel model = CorePlugin.modelPersistence().loadModel(workspaceProject);
-        if (model == null) {
-            model = PersistentModelFactory.from(workspaceProject, new Path("build"), ImmutableList.<IPath>of(), ImmutableList.<IClasspathEntry>of(), ImmutableList.<IPath>of(), ImmutableList.<IPath>of());
+        if (!model.isPresent()) {
```

Comment:
The problem is that the PersistenceModelBuilder is initialized from the PersistenceModel passed as an argument. My gut feeling is that it's better to have a centralized default model passed to the updates.